src/app/control-planes/sources.ts

```diff
@@ -1,20 +1,26 @@
-import { DataSourceResponse } from '@/app/application/services/data-source/DataSourcePool'
+import type { DataSourceResponse } from '@/app/application'
 import type Env from '@/services/env/Env'
+import type KumaApi from '@/services/kuma-api/KumaApi'
+import type { GlobalInsight } from '@/types/index.d'
 
 export type ControlPlaneAddresses = {
   http: string
   kds: string
 }
 
 export type ControlPlaneAddressesSource = DataSourceResponse<ControlPlaneAddresses>
-export const sources = (env: Env['var']) => {
+export type GlobalInsightSource = DataSourceResponse<GlobalInsight>
+
+export const sources = (env: Env['var'], api: KumaApi) => {
   return {
-    '/control-plane/addresses': async (_params: {}, source: { close: () => void }): Promise<ControlPlaneAddresses> => {
-      source.close()
```

Comment:
We recently removed the need to close sources if you are returning a straight forward promise, so I removed all this as I was here.